app/FileJson.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace ProjektGopher\Whisky;
+
+use Illuminate\Support\Facades\File;
+use Swaggest\JsonSchema\Context;
+use Swaggest\JsonSchema\Schema;
+
+class FileJson
+{
+    public string $path;
+
+    public function __construct(
+        public string $filePath,
+    ) {
+        $this->path = $filePath;
+    }
+
+    public static function make(string $filePath): FileJson
+    {
+        return new FileJson($filePath);
+    }
+
+    public function read(bool $validate = true): string|array|null
+    {
+        $content = json_decode(File::get($this->path), true);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            $msg = 'Invalid JSON';
+            if (function_exists('json_last_error_msg')) {
+                $msg .= ': '.json_last_error_msg();
+                $msg .= ' in '.$this->path;
+            }
+            throw new \Exception($msg);
+        }
+
+        if ($validate) {
+            try {
+                $contentJson = json_decode(File::get($this->path));
+                $this->validate($contentJson);
+            } catch (\Exception $e) {
+                $msg = 'Invalid JSON schema';
+                $msg .= ': '.$e->getMessage();
+                $msg .= ' in '.$this->path;
+                throw new \Exception($msg);
+            }
+        }
+
+        return $content;
+    }
```

Comment:
```suggestion
    public function read(bool $validate = true): string|array|null
    {
        if (json_last_error() !== JSON_ERROR_NONE) {
            $msg = 'Invalid JSON';
            if (function_exists('json_last_error_msg')) {
                $msg .= ': '.json_last_error_msg();
                $msg .= ' in '.$this->path;
            }
            throw new \Exception($msg);
        }

        if ($validate) {
            try {
                $contentJson = json_decode(File::get($this->path));
                $this->validate($contentJson);
            } catch (\Exception $e) {
                $msg = 'Invalid JSON schema';
                $msg .= ': '.$e->getMessage();
                $msg .= ' in '.$this->path;
                throw new \Exception($msg);
            }
        }

        return json_decode(File::get($this->path), true);
    }
```

Reply:
I first validate the `json_last_error` function but `json_decode` must have been executed before. Then I do the validation with the schema library.
Note that the `read` function returns an "array", so it returns `$content` with `json_decode($string, true)`, while schema validation needs a "stdClass" so `$contentJson` is with `json_decode($string, false)`.